src/CoreSchema.F90

```diff
@@ -268,38 +271,39 @@ end function matches_exponent
     
   end function matches_real
 
-  logical function to_logical(text)
+  logical function to_logical(text,rc)
     character(*), intent(in) :: text
+    integer, optional, intent(out) :: rc
```

Comment:
The `SELECT CASE` construct needs a `DEFAULT` clause which will signal a failure.  I'd suggest a change, but cannot do it here because it is not in the lines that were modified in the PR.